### PR TITLE
fix: handle errors correctly

### PR DESCRIPTION
**Pull request by MatousJobanek**
There were two cases when the errors weren't properly handled:
https://github.com/codeready-toolchain/host-operator/blob/master/pkg/configuration/configuration.go#L88
https://github.com/codeready-toolchain/host-operator/blob/master/pkg/controller/masteruserrecord/sync.go#L121

As part of this PR I added tests that cover the error cases

### CODE REVIEW

File: pkg/controller/masteruserrecord/sync_test.go [hunk 1]
```diff
@@ -159,24 +149,45 @@ func TestSynchronizeSpec(t *testing.T) {
 
 func TestSynchronizeStatus(t *testing.T) {
 	// given
-	logf.SetLogger(logf.ZapLogger(true))
```
Comment (xcoulon) [inline pkg/controller/masteruserrecord/sync_test.go:162]: oh, why did you remove these lines? I find it useful to have the `INFO` log messages.
(It's not a requirement to revert the change, though)

File: pkg/controller/masteruserrecord/sync_test.go [hunk 2]
```diff
@@ -243,8 +300,48 @@ func TestSyncMurStatusWithUserAccountStatusWhenCompleted(t *testing.T) {
...
+		// given
+		hostClient := test.NewFakeClient(t, mur)
+		sync, memberClient := prepareSynchronizer(t, userAccount, mur, hostClient)
+
+		// when
+		err := sync.synchronizeStatus()
+
+		// then
+		require.NoError(t, err)
+		verifySyncMurStatusWithUserAccountStatus(t, memberClient, hostClient, userAccount, mur, toBeProvisionedNotificationCreated(), toBeProvisionedNotificationCreated())
```
Comment (xcoulon) [inline pkg/controller/masteruserrecord/sync_test.go:313]: just curious, why to you need `toBeProvisionedNotificationCreated()` twice?

File: pkg/controller/masteruserrecord/sync_test.go (hunk 1 above)
Comment (MatousJobanek) [reply pkg/controller/masteruserrecord/sync_test.go:162]: because I moved it to `prepareSynchronizer` function - it was there before, so it was just a duplication:
https://github.com/codeready-toolchain/host-operator/blob/5361b85e3114bbb7972a88536f0486c159b7c9b4/pkg/controller/masteruserrecord/sync_test.go#L641-L649

File: pkg/controller/masteruserrecord/sync_test.go (hunk 2 above)
Comment (MatousJobanek) [reply pkg/controller/masteruserrecord/sync_test.go]: wrong copy-paste :man_facepalming: 
good catch, thanks - fixed in 2368e4f

File: pkg/controller/masteruserrecord/sync_test.go [hunk 3]
```diff
@@ -159,24 +149,45 @@ func TestSynchronizeSpec(t *testing.T) {
...
+		hostClient.MockStatusUpdate = func(ctx context.Context, obj runtime.Object, opts ...client.UpdateOption) error {
+			return fmt.Errorf("some error")
+		}
+		sync, _ := prepareSynchronizer(t, userAccount, mur, hostClient)
+
+		// when
+		err := sync.synchronizeStatus()
+
+		// then
+		require.Error(t, err)
```
Comment (tinakurian) [inline pkg/controller/masteruserrecord/sync_test.go:184]: maybe we should also check the error itself?

File: pkg/configuration/configuration_test.go [hunk 4]
```diff
@@ -82,15 +87,32 @@ func TestLoadFromSecret(t *testing.T) {
...
+		restore := test.SetEnvVarAndRestore(t, "HOST_OPERATOR_SECRET_NAME", "test-secret")
+		defer restore()
+
+		cl := test.NewFakeClient(t)
+
+		// when
+		config, err := configuration.LoadConfig(cl)
+
+		// then
+		require.Error(t, err)
```
Comment (tinakurian) [inline pkg/configuration/configuration_test.go:113]: maybe we should check the error message here as well?

File: pkg/controller/masteruserrecord/sync_test.go (hunk 3 above)
Comment (MatousJobanek) [reply pkg/controller/masteruserrecord/sync_test.go:184]: added 1a7b7de

File: pkg/configuration/configuration_test.go (hunk 4 above)
Comment (MatousJobanek) [reply pkg/configuration/configuration_test.go:113]: added 1a7b7de
